src/d/d_wood.cpp

```diff
@@ -4,184 +4,899 @@
+      scaledNormZ = -gndNorm.z * unkFloat;
+    } else {
+      scaledNormY = 0.0f;
+      scaledNormZ = 0.0f;
+    }
+
+    MtxP mtx = mDoMtx_stack_c::get();
+
+    mtx[0][0] = unkFloat;
+    mtx[0][1] = gndNorm.x;
+    mtx[0][2] = 0.0f;
+    mtx[0][3] = mPos.x;
+
+    mtx[1][0] = -gndNorm.x * scaledNormY;
+    mtx[1][1] = gndNorm.y;
+    mtx[1][2] = scaledNormZ;
+    mtx[1][3] = kGroundHeightBias + gndHeight;
+
+    mtx[2][0] = gndNorm.x * scaledNormZ;
+    mtx[2][1] = gndNorm.z;
+    mtx[2][2] = scaledNormY;
+    mtx[2][3] = mPos.z;
+    mDoMtx_stack_c::scaleM(L_attr.kUncutShadowScale, kGroundHeightBias,
+                           L_attr.kUncutShadowScale);
+    mDoMtx_copy(mDoMtx_stack_c::now, mShadowModelMtx);
+    return true;
+  } else {
+    return false;
+  }
 }
 
 /* 800BE93C-800BEA28       .text set_mtx__Q25dWood6Unit_cFPQ25dWood5Anm_c */
-void dWood::Unit_c::set_mtx(dWood::Anm_c*) {
-    /* Nonmatching */
-}
+void dWood::Unit_c::set_mtx(dWood::Anm_c *anim) {
+  int iVar1;
 
-/* 800BEA28-800BEA50       .text clear__Q25dWood6Unit_cFv */
-void dWood::Unit_c::clear() {
-    /* Nonmatching */
-}
+  iVar1 = mAnmIdx;
 
-/* 800BEA50-800BEE9C       .text cc_hit_before_cut__Q25dWood6Unit_cFPQ25dWood8Packet_c */
-void dWood::Unit_c::cc_hit_before_cut(dWood::Packet_c*) {
-    /* Nonmatching */
-}
+  mDoMtx_copy(anim[iVar1].mModelMtx, mDoMtx_stack_c::get());
+  mDoMtx_stack_c::now[0][3] = mDoMtx_stack_c::now[0][3] + (mPos).x;
+  mDoMtx_stack_c::now[1][3] = mDoMtx_stack_c::now[1][3] + (mPos).y;
+  mDoMtx_stack_c::now[2][3] = mDoMtx_stack_c::now[2][3] + (mPos).z;
+  mDoMtx_concat(j3dSys.getViewMtx(), mDoMtx_stack_c::now, mModelViewMtx);
+
+  mDoMtx_copy(anim[iVar1].mTrunkModelMtx, mDoMtx_stack_c::get());
+  mDoMtx_stack_c::now[0][3] = (mPos).x;
+  mDoMtx_stack_c::now[1][3] = (mPos).y;
+  mDoMtx_stack_c::now[2][3] = (mPos).z;
+  mDoMtx_concat(j3dSys.getViewMtx(), mDoMtx_stack_c::get(),
+                mTrunkModelViewMtx);
 
-/* 800BEE9C-800BEEA0       .text cc_hit_after_cut__Q25dWood6Unit_cFPQ25dWood8Packet_c */
-void dWood::Unit_c::cc_hit_after_cut(dWood::Packet_c*) {
-    /* Nonmatching */
+  mDoMtx_concat(j3dSys.getViewMtx(), mShadowModelMtx,
+                mShadowModelViewMtx);
 }
 
+/* 800BEA28-800BEA50       .text clear__Q25dWood6Unit_cFv */
+void dWood::Unit_c::clear() { cLib_memSet(this, 0, 0x18c); }
+
+/* 800BEA50-800BEE9C       .text
+ * cc_hit_before_cut__Q25dWood6Unit_cFPQ25dWood8Packet_c */
+void dWood::Unit_c::cc_hit_before_cut(dWood::Packet_c *packet) {
+  int animIdx;
+  int oldAnimIdx;
+
+  dCcMassS_HitInf inf;
+  fopAc_ac_c *actor;
+  u32 ret = dComIfG_Ccsp()->ChkMass(&mPos, &actor, &inf);
+
+  if (mAnimCooldown > 0) {
+    mAnimCooldown -= 1;
+  }
+
+  // Evaluate for attacks that will not cut us down
+  if ((ret & 1)) {
+    cCcD_Obj *atHitObj = inf.GetAtHitObj();
+    if (atHitObj != NULL &&
+        // @TODO: I think the cCcD_ObjAtType enum is reversed. E.g. the first
+        // one should be Machete (bit 21) not Wind
```

Comment:
the enum definitely isn't reversed. the machete actor is using AT_TYPE_MACHETE and the fan actor is using AT_TYPE_WIND currently. why do you believe it's backwards?

Reply:
Ah, I was wrong, old comment